APMrover2/mode.h

```diff
@@ -280,6 +280,51 @@ class ModeGuided : public Mode
     float _desired_yaw_rate_cds;// target turn rate centi-degrees per second
 };
 
+class ModeBreadcrumb : public ModeGuided
+{
+public:
+
+    uint32_t mode_number() const override { return BREADCRUMB; }
+    const char *name4() const override { return "BREADCRUMB"; }
+
+    // methods that affect movement of the vehicle in this mode
+    void update() override;
+
+    // attributes of the mode
+    bool is_autopilot_mode() const override { return true; }
+
+    void mavlink_packet_received(const mavlink_message_t &msg);
+
+
+protected:
+
+    bool _enter() override;
+
+    void _exit()  override;
+
+private:
+
+    uint8_t target_srcid = 2; //#TODO maybe make a user parameter?
```

Comment:
I'd like to get this merged, so I wouldn't hold the PR up for adding a parameter here.  But it really has to be done to make the mode useful.